compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -6200,7 +6200,12 @@ void solveSelectTypeAndResolveType(BLangQueryExpr queryExpr, BLangExpression sel
             case TypeTags.NONE:
             default:
                 // contextually expected type not given (i.e var).
-                selectType = checkExpr(selectExp, env, type, data);
+                selectType = checkExprSilent(nodeCloner.cloneNode(selectExp), type, data);
```

Comment:
Even with the following example, we are still getting errors similar to 'incompatible types: expected 'xml', found 'string
```ballerina
type Person record {|
    string firstName;
    string lastName;
    int age;
|};

function testFunc() {
    Person p1 = {firstName: "Alex", lastName: "George", age: 33};
    Person p2 = {firstName: "John", lastName: "David", age: 35};
    Person p3 = {firstName: "Max", lastName: "Gomaz", age: 33};
    Person[] personList = [p1, p2, p3];

    xml outputPersonList =
            from Person person in personList
            let int newAge = 20
            where person.age == 33
            select person.firstName;
}
```